pkg/maps/egressmap/policy.go

```diff
@@ -201,11 +218,12 @@ func (v *EgressPolicyVal4) String() string {
 // IP, destination CIDR) tuple.
 func (m *policyMap) Lookup(sourceIP netip.Addr, destCIDR netip.Prefix) (*EgressPolicyVal4, error) {
 	key := NewEgressPolicyKey4(sourceIP, destCIDR)
-	val := EgressPolicyVal4{}
-
-	err := m.m.Lookup(&key, &val)
+	val, err := m.m.Lookup(&key)
+	if err != nil || val == nil {
```

Comment:
Why does `val == nil` need to be checked?

Reply:
Actually, I referenced other usecases, e.g. https://github.com/cilium/cilium/blob/980c5dd0db7e3d882490a98f7451c1f34bd9a72f/pkg/maps/ctmap/lookup.go#L156-L161

I think this check is intended to prevent the type assertion at line 226 from panicking

Reply:
Doesn't that mean that we can get `nil, nil` return value from this method? Do callers deal with that appropriately?

Reply:
in practice we don't even use that method, beside in unit tests, as we always iterate the full map with `IterateWithCallback` (now renamed to `DumpWithCallback`) so should be fine

Reply:
> Doesn't that mean that we can get nil, nil return value from this method? 

As far as I looked into the `Lookup` implementation, it creates a pointer using `New()` and unmarshals a buffer into the provided value. So we might not get `nil, nil`, but I'm not 100% sure.

Reply:
> in practice we don't even use that method, beside in unit tests

The code is not unit test specific so someone might end up calling it in the future.

> So we might not get nil, nil, but I'm not 100% sure.

Ok, so if `m.m.Lookup` never returns `nil, nil` then to me it makes sense to drop `v == nil` check in this function.

If we really need to guard against `nil`ness a fallible type assertion like `value, ok := v.(*CtEntry)` would be better imo (`ok` becomes false if `v` is `nil`).

Reply:
I prefer using the fallible type assertion because there isn't any guarantee that it never return `nil nil` in the future. Does it make sense?

Reply:
I think you don't have to be so cautious, since going to `nil, nil` is not a common change to make. If you prefer the type assertion you can still go for it.

Reply:
OK, I will just drop the nil check then. Thank you for the feedback!